src/main/java/seedu/address/logic/commands/RequiredCommand.java

```diff
@@ -0,0 +1,319 @@
+import javafx.collections.ObservableList;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyGradPad;
+import seedu.address.model.module.Module;
+import seedu.address.storage.JsonGradPadStorage;
+
+public class RequiredCommand extends Command {
+    public static final String COMMAND_WORD = "required";
+    public static final String MESSAGE_FOUNDATION = "These are the Foundation Modules you have yet to take: ";
+    public static final String MESSAGE_FAILURE_FOUNDATION = "There was an error loading the Foundation Modules :(";
+    public static final String MESSAGE_FAILURE_SCI = "There was an error loading the Science Modules :(";
+    public static final String MESSAGE_SCIENCE = "You have not completed your Science Module requirement,"
+            + " use the 'science' command to view the available modules.";
+    public static final String MESSAGE_FAILURE_INTERN = "There was an error loading the Internship Modules :(";
+    public static final String MESSAGE_INTERN_1 = "You have yet to complete your 12MCs worth of Internship Modules.";
+    public static final String MESSAGE_INTERN_2 = "These are the Internship Modules you can take:";
+    public static final String MESSAGE_SUCCESS_FOUNDATION = "You have completed all your Foundation Modules!";
+    public static final String MESSAGE_SUCCESS_SCIENCE = "You have completed your Science Module Requirement!";
+    public static final String MESSAGE_SUCCESS_INTERN = "You have completed your Internship Module Requirement!";
+    public static final String MESSAGE_ITPROF = "These are the IT Professionalism Modules you have yet to take:";
+    public static final String MESSAGE_SUCCESS_ITPROF = "You have completed your "
+            + "IT Professionalism Module Requirement!";
+    public static final String MESSAGE_MATHANDSCI = "These are the Math and Science Modules you have yet to take:";
+    public static final String MESSAGE_SUCCESS_MATHANDSCI = "You have completed your "
+            + "Math and Science Module Requirement!";
+    public static final String MESSAGE_FAILURE_ITPROF = "There was an error loading the IT Professionalism Modules :(";
+    public static final String MESSAGE_FAILURE_MATHANDSCI = "There was an error loading"
+            + " the Math and Science Modules :(";
+    public static final Path FOUNDATION_PATH = Paths.get("src/main/data/foundationmodules.json");
+    public static final Path INTERNSHIP_PATH = Paths.get("src/main/data/industrialexperience.json");
+    public static final Path ITPROF_PATH = Paths.get("src/main/data/ITProfessionalism.json");
+    public static final Path MATHANDSCI_PATH = Paths.get("src/main/data/mathandsciencemodules.json");
+    public static final Path SCIENCE_PATH = Paths.get("src/main/data/sciencemodules.json");
+
+    private ObservableList<Module> modules;
+    private String leftOverModules = "";
+    private Optional<ReadOnlyGradPad> foundationStorage;
```

Comment:
Maybe we can initialize this to an empty `Optional` instead of leaving it as null? I think it's a bit weird that the field can either be `null`, empty `Optional`, or filled `Optional` when the purpose of Optionals is to abstract away nulls anyway HAHA 

But this one is really a minor suggestion :grin:

![Screenshot 2020-10-22 at 5 05 16 PM](https://user-images.githubusercontent.com/31149701/96849889-c51c3500-1488-11eb-99f2-4435012efa48.png)

Reply:
Changed the whole class design so, no more Optional<> attributes HAHAHA